src/components/datetime.rs

```diff
@@ -18,9 +18,68 @@ use tinystr::TinyAsciiStr;
 use super::{
     calendar::{CalendarDateLike, GetTemporalCalendar},
     duration::normalized::{NormalizedTimeDuration, RelativeRoundResult},
-    Date, Duration, Time,
+    Date, Duration, MonthCode, PartialDate, PartialTime, Time,
 };
 
+/// A partial DateTime record
+pub struct PartialDateTime {
+    date: PartialDate,
+    time: PartialTime,
+}
```

Comment:
Maybe we should just expose the fields as public? having a big method initializer with a lot of parameters seems overkill for essentially an "options" struct that is only passed to an initializer method. It would also make it really easy to set only a single field, since we could call `default` then only initialize the fields we want.

Reply:
Yeah, I think that makes sense. I initially went with the above to avoid the multiple parameters of `from_parts`, but then ended up adding it anyways.

Reply:
I went ahead and applied that to all the partials currently completed. starting out with a lot of simplicity is probably the best way to handle it.